File: app/src/main/java/net/tlalka/android/fiszki/view/adapters/LanguageAdapter.java
package net.tlalka.android.fiszki.view.adapters;

import android.content.Context;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import net.tlalka.android.fiszki.R;
import net.tlalka.android.fiszki.model.types.LanguageType;

import java.util.List;

public class LanguageAdapter extends AbstractAdapter<LanguageType> {

    public LanguageAdapter(Context context, List<LanguageType> languageTypes) {
        super(context, languageTypes);
    }

    @Override
    public View getView(int position, View convertView, ViewGroup viewGroup) {
        if (convertView == null) {
            convertView = super.getLayoutInflater().inflate(R.layout.settings_spinner_item, viewGroup, false);
            convertView.setTag(new ViewHolderPattern(convertView));
        }

        ViewHolderPattern viewHolderPattern = (ViewHolderPattern) convertView.getTag();
        viewHolderPattern.item.setText(super.getItem(position).name());
        return convertView;
    }

    private static class ViewHolderPattern {
        public final TextView item;

        public ViewHolderPattern(View view) {
            this.item = (TextView) view.findViewById(R.id.spinner_item);
        }
    }
}

File: app/src/main/kotlin/net/tlalka/fiszki/view/adapters/LanguageAdapter.kt
package net.tlalka.fiszki.view.adapters

import android.content.Context
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import net.tlalka.fiszki.R
import net.tlalka.fiszki.model.types.LanguageType

class LanguageAdapter(context: Context, languageTypes: List<LanguageType>)
        : AbstractAdapter<LanguageType>(context, languageTypes) {

    override fun getView(position: Int, convertView: View?, viewGroup: ViewGroup?): View {
        val itemView = super.getItemView(convertView, viewGroup, R.layout.settings_spinner_item)
        itemView.tag = itemView.tag ?: ViewHolderPattern(itemView)

        val viewHolderPattern = itemView.tag as ViewHolderPattern
        viewHolderPattern.item.text = super.getItem(position).name
        return itemView
    }

    private class ViewHolderPattern(view: View) {
        val item: TextView = view.findViewById(R.id.settings_spinner_item)
    }
}
